core/src/plugins/filed/BareosFdPluginLibcloud.py

```diff
@@ -0,0 +1,386 @@
+                        self.options["temporary_download_directory"] = value
+                    else:
+                        self.options[option_map[option]] = value
+                except:
+                    debugmessage(
+                        100,
+                        "Could not evaluate: %s in config file %s"
+                        % (value, config_filename),
+                    )
+                    return False
+
+        return True
+
+    def start_backup_job(self, context):
+        jobmessage(
+            "M_INFO",
+            "Start backup, try to connect to %s:%s"
+            % (self.options["host"], self.options["port"]),
+        )
+
+        if BareosLibcloudApi.probe_driver(self.options) == "failed":
+            jobmessage(
+                "M_FATAL",
+                "Could not connect to libcloud driver: %s:%s"
+                % (self.options["host"], self.options["port"]),
+            )
+            return bRCs["bRC_Error"]
+
+        jobmessage(
+            "M_INFO",
+            "Connected, last backup: %s (ts: %s)" % (self.last_run, self.since),
+        )
+
+        self.api = None
+        try:
+            self.api = BareosLibcloudApi(
+                self.options, self.last_run, self.options["temporary_download_directory"]
+            )
+            debugmessage(100, "BareosLibcloudApi started")
+        except Exception as e:
+            debugmessage(100, "Error: %s" % e)
+            jobmessage("M_FATAL", "Something went wrong with BareosLibcloudApi: %s" % e)
+            return bRCs["bRC_Cancel"]
+
+        return bRCs["bRC_OK"]
+
+    def end_backup_job(self, context):
+        if self.active:
+            self.__shutdown()
+        return bRCs["bRC_OK"]
+
+    def check_file(self, context, fname):
+        # All existing files/objects are passed to bareos
+        # If bareos have not seen one, it does not exists anymore
+        return bRCs["bRC_Error"]
+
+    def __shutdown(self):
+        self.active = False
+        jobmessage(
+            "M_INFO",
+            "BareosFdPluginLibcloud finished with %d files"
+            % (self.number_of_objects_to_backup)
+        )
+
+        debugmessage(100, "Shut down BareosLibcloudApi")
+        self.api.shutdown()
+        debugmessage(100, "BareosLibcloudApi is shut down")
+
+    def start_backup_file(self, context, savepkt):
+        error = False
+        while self.active:
+            if self.api.check_worker_messages() != SUCCESS:
+                self.active = False
+                error = True
+            else:
+                self.current_backup_job = self.api.get_next_job()
+                if self.current_backup_job != None:
+                    break
+                elif self.api.worker_ready():
+                    self.active = False
+                else:
+                    sleep(0.01)
+
+        if not self.active:
+            self.__shutdown()
+            savepkt.fname = "empty"  # dummy value
```

Comment:
can this collide with a real file named "emtpy"? can the fname not be empty ("") when being skipped?